From abc886777b36ec1477ca7586ac41e4788401bf22 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Fri, 21 Mar 2014 18:40:25 +0800
Subject: [PATCH 0779/1074] ENGR00304676-3: pci: host: imx6: remove gpc
 operate code

remove the code about gpc, since it will be done in gpc driver.
That makes driver more clean and better.

Signed-off-by: Robin Gong <b38343@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 drivers/pci/host/pci-imx6.c |   12 ------------
 1 files changed, 0 insertions(+), 12 deletions(-)

diff --git a/drivers/pci/host/pci-imx6.c b/drivers/pci/host/pci-imx6.c
index 5cb685b..ab75674 100644
--- a/drivers/pci/host/pci-imx6.c
+++ b/drivers/pci/host/pci-imx6.c
@@ -67,7 +67,6 @@ struct imx6_pcie {
 	struct clk		*dis_axi;
 	struct pcie_port	pp;
 	struct regmap		*iomuxc_gpr;
-	struct regmap		*gpc_ips_reg;
 	struct regulator	*pcie_regulator;
 	void __iomem		*mem_base;
 };
@@ -356,8 +355,6 @@ static void imx6_pcie_init_phy(struct pcie_port *pp)
 		if (ret)
 			dev_info(pp->dev, "failed to enable pcie regulator.\n");
 
-		/* Disable PCIe isolation, APIS1_GPC_IPS 0x020D_C000 offset 0 */
-		regmap_update_bits(imx6_pcie->gpc_ips_reg, 0, BIT(7), 1 << 7);
 	}
 	regmap_update_bits(imx6_pcie->iomuxc_gpr, IOMUXC_GPR12,
 			IMX6Q_GPR12_PCIE_CTL_2, 0 << 10);
@@ -896,15 +893,6 @@ static int __init imx6_pcie_probe(struct platform_device *pdev)
 			return ret;
 		}
 
-		/* Grab GPC IPS config register range */
-		imx6_pcie->gpc_ips_reg =
-			 syscon_regmap_lookup_by_compatible("fsl,imx6q-gpc");
-		if (IS_ERR(imx6_pcie->gpc_ips_reg)) {
-			dev_err(&pdev->dev, "unable to find gpc ips registers\n");
-			ret = PTR_ERR(imx6_pcie->gpc_ips_reg);
-			return ret;
-		}
-
 		/* Get pcie regulator */
 		imx6_pcie->pcie_regulator = devm_regulator_get(pp->dev, "pcie");
 
-- 
1.7.5.4

